### PR TITLE
avoid #unused# leaking into ArgumentError message

### LINKED ISSUE #16536: name `#unused#` leaking into method definition error message
it would be preferably here to detect that the argument name is a special symbol and use the argument number instead.
```julia
julia> f(::1) = 2
ERROR: ArgumentError: invalid type for argument `#unused#` in method definition for f at REPL[1]:1
 in eval(::Module, ::Any) at ./boot.jl:225
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at event.jl:92
```

### PR DESCRIPTION
**Pull request by vtjnash**
fix #16536

### CODE REVIEW

File: src/toplevel.c [hunk 1]
```diff
@@ -766,15 +766,25 @@ JL_DLLEXPORT void jl_method_def(jl_svec_t *argdata, jl_lambda_info_t *f, jl_valu
     f = m->lambda_template; // because jl_new_method makes a copy
     jl_check_static_parameter_conflicts(m, tvars);
 
-    // TODO
```
Comment (vtjnash) [inline src/toplevel.c]: @JeffBezanson added this in afd4eb038dcd9fa3e512509f4c332a9e7763ba59, but I don't see anything that seems to need to-be-done. I'm guessing this is just out-of-date and was already fixed in the same PR?

**Comment by tkelman** [comment]
please add a test

**Comment by vtjnash** [comment]
any idea where the other tests are for this error?

**Comment by tkelman** [comment]
not positive, but there was a good set of tests added recently for kwarg errors

**Comment by yuyichao** [comment]
Replutil

**Pull request merged**